### PR TITLE
Public ELB

### PR DESCRIPTION
**Pull request by myyk**
These changes will let us launch an ELB that is publicly available in addition to our private ELB. They both will point at the same autoscale group.

@jbronn @hobbeswalsh @cyc cc @sbuss (Ah ha, I could still cc you when you're gone on the publicly available code!)

### CODE REVIEW

**Comment by sbuss** [comment]
LGTM, but might want to add some docs?

**Comment by myyk** [comment]
I was thinking about this some more. I might want to make some changes so that if `VPC_PUBLIC_ELB_SUBNETS` is empty, that we don't create the public ELB. Since we will want a way to not create a public ELB for the Dev cluster, I think.

**Comment by hobbeswalsh** [comment]
> we don't create the public ELB. Since we will want a way to not create a public ELB for the Dev cluster, I think.

Yeah, either this or a flag to toggle it. :+1:

**Comment by myyk** [comment]
@greglook Can you take a look at the CloudFormation changes?

File: contrib/aws/deis.template.json [hunk 1]
```diff
@@ -303,6 +322,68 @@
         ]
       }
     },
+    "DeisPublicWebELB": {
+      "Type": "AWS::ElasticLoadBalancing::LoadBalancer",
+      "Condition": "UsePublicELB",
+      "DependsOn" : "GatewayToInternet",
```
Comment (greglook) [inline contrib/aws/deis.template.json:328]: Hmm, I thought this needed to be an array of logical names. Okay if this works though.

File: contrib/aws/deis.template.json [hunk 2]
```diff
@@ -303,6 +322,68 @@
...
+          {
+            "InstancePort": "443",
+            "InstanceProtocol": "TCP",
+            "LoadBalancerPort": "443",
+            "Protocol": "TCP"
+          }
+        ],
+        "SecurityGroups": [
+          {
+            "Fn::GetAtt": ["DeisPublicWebELBSecurityGroup", "GroupId"]
```
Comment (greglook) [inline contrib/aws/deis.template.json]: Does a standard `{"Ref": "DeisPublicWebELBSecurityGroup"}` not work here?

File: contrib/aws/deis.template.json [hunk 3]
```diff
@@ -399,6 +495,14 @@
     "DNSName" : {
       "Description" : "DNS Name of the ELB",
       "Value" :  { "Fn::GetAtt" : ["DeisWebELB", "DNSName"] }
-    }
+    },
+    "PublicDNSName": { "Fn::If": [
```
Comment (greglook) [inline contrib/aws/deis.template.json]: You don't need the `Fn::If` here, you can just add a `Condition` field to this output.

Comment (myyk) [inline contrib/aws/deis.template.json]: Oh, nice.

File: contrib/aws/deis.template.json (hunk 1 above)
Comment (myyk) [inline contrib/aws/deis.template.json:328]: Idk, this is copied from their DeisWebELB. Let's leave it as is.

File: contrib/aws/deis.template.json (hunk 2 above)
Comment (myyk) [inline contrib/aws/deis.template.json]: Oh, I think it would. Fixed.

**Comment by myyk** [comment]
LGTM from @greglook. Other issues look like stalemate. I'm merging so that I can continue. (Willing to iterate if the debate decides we should make them.)

**Pull request merged**